src/axelrod_dojo/archetypes/gambler.py

```diff
@@ -55,6 +66,26 @@ def create_vector_bounds(self):
 
         return lb, ub
 
+    @staticmethod
+    def mutate_pattern(pattern, mutation_probability):
+        for i, _ in enumerate(pattern):
+            if random.random() <= mutation_probability:
+                pattern[i] = random.random()
+        return pattern
+
+    def mutate(self):
+        self.pattern = self.mutate_pattern(self.pattern, self.mutation_probability)
+
+    def crossover(self, other):
+        pattern1 = self.pattern
+        pattern2 = other.pattern
+
+        midpoint = int(random.randint(0, len(pattern1)) / 2)
```

Comment:
Why not `cross_point = random.randint(0, len(pattern1))` or the procedure as FSM [here](https://github.com/Axelrod-Python/axelrod-dojo/blob/master/src/axelrod_dojo/archetypes/fsm.py#L93)?

I assume the patterns are the same length, otherwise there are edge cases where you could choose beyond the end of `pattern2`. If so let's add a comment that states the assumption, or raise an exception if the assumption doesn't hold.